### PR TITLE
Fix RIPPLE-93. Allow proper local emulation independently from the current working directory.

### PR DESCRIPTION
**Pull request by busykai**
This PR consists of 3 commits, each addressing its own specific issue (see commit message for details) to be able to address the issue described in [RIPPLE-93](https://issues.apache.org/jira/browse/RIPPLE-93) properly.

CC: @TimBarham

### CODE REVIEW

File: lib/cli/emulate.js [hunk 1]
```diff
@@ -20,11 +20,35 @@
...
     call: function (opts) {
-        self.start(opts);
+        if (self.validateOptions(opts)) {
+            self.start(opts);
+        } else {
+            console.error(('Options not understood: ' + process.argv.splice(2).join(' ')).red.bold);
+            help.call('emulate');
+        }
+    },
+    validateOptions: function (opts) {
```
Comment (TimBarham) [inline lib/cli/emulate.js]: Is there a reason this is an exported method and not a private function?

**Comment by busykai** [comment]
RIPPLE-93?

**Comment by TimBarham** [comment]
Sorry, that question was specifically for commit [4cfa798](https://github.com/busykai/incubator-ripple/commit/4cfa79801d56cf75a304cde4ddda9fe90e0555fc).

**Comment by busykai** [comment]
It removes any logic from the module load-time. Say, if you specified incorrect cmd-line arguments, it would still execute and potentially print a warning. Instead, it's moved to the time the module is actually being used (instead of just loaded).

File: lib/cli/emulate.js (hunk 1 above)
Comment (busykai) [inline lib/cli/emulate.js]: I was thinking that the command-line processing would actually use it before calling start, but that would require implementing the same functionality in other modules which, in turn, would contaminate this PR with irrelevant changes. Do you want me to make it private? Or, I could finish global command-line option validation and handling.

Comment (busykai) [inline lib/cli/emulate.js]: What I mean we could leave it here and I would submit another PR which handles the options better.

**Comment by TimBarham** [comment]
Ah, ok, makes sense.

File: lib/cli/emulate.js (hunk 1 above)
Comment (TimBarham) [inline lib/cli/emulate.js]: Yeah, that's fine. As long as you have a goal to use this externally, I'm fine merging it as is.
